app/main/views/services.py

```diff
@@ -766,7 +767,10 @@ def previous_services(framework_slug, lot_slug):
             # Don't copy a service if the lot has a one service limit and the supplier already has a draft for that lot
             drafts, complete_drafts = get_lot_drafts(data_api_client, framework_slug, lot_slug)
             if drafts or complete_drafts:
-                abort(400, f"You already have a draft {lot['name'].lower()} service.")
+                return render_error_page(
```

Comment:
My understanding of how the error handling works is limited. Why do you need to call render_error_page here? If you left it as it was, wouldn't the error handler registered in flask_init pick this up? You might need to add a `error_message` kwarg to the csrf handler which could be passed through to `render_error_page` there?

Reply:
Yes, exactly - 400 handling is less than ideal as there's a bug in Flask 0.10.x which means CSRFErrors are treated by error handlers as ordinary 400s. To get round this, in `dmutils.errors.csrf_handler` we listen for *all* 400s then send any that aren't CSRFs to the usual 400 template (no error message there yet, unfortunately). 

 We have a tech debt ticket to sort this out once we've upgraded Flask: https://trello.com/c/wb3WqzEY/545-handle-csrferror-exceptions-separately-from-400-errors

Reply:
If we call `abort(400)` the error message will be lost.